lamindb/curators/__init__.py

```diff
@@ -380,13 +380,13 @@ def _check_valid_keys(self, extra: set | None = None) -> None:
                 raise TypeError(f"{name} must be a dictionary!")
             valid_keys = set(self._df.columns) | {"columns"} | extra
             nonval_keys = [key for key in d.keys() if key not in valid_keys]
-            n = len(nonval_keys)
-            s = "s" if n > 1 else ""
-            are = "are" if n > 1 else "is"
-            if len(nonval_keys) > 0:
-                raise ValidationError(
-                    f"key{s} passed to {name} {are} not present in columns: {colors.yellow(_format_values(nonval_keys))}"
-                )
+            len(nonval_keys)
+            # throwing an error here is problematic because my dataset might have missing columns
+            # and I still want to define sources mappings on the Curator level
```

Comment:
I don't know who wrote this, so FYI @Zethson @sunnyosun -- I don't think this makes sense for the reason defined in the comment

We definitely want to be able to pass datasets with missing columns to the Curator constructor without raising an error